context/AuthContextProvider.tsx

```diff
@@ -87,7 +89,9 @@ export const AuthContextProvider = ({
    */
   const getUser = async (): Promise<IUser | undefined> => {
     if (!isSessionInLocalStorage()) {
-      router.push('/login');
+      if (pathname !== '/register') {
+        router.push('/login');
```

Comment:
Curious as to why it's another if stmt? Could we do an or: `!isSessionInLocalStorage() || pathname !== '/register'`?

Reply:
if i combined them as that then itll constantly run if the pathName isn't /register, which we don't want that. We first want to check that there is !isSessionInLocalStorage() and only then will we check if the url doesn't have /register then itll push the url to /login.

Reply:
When you state `itll constantly run`, is `it` the webpage we are referring to? Just to make sure I understand.

Reply:
ya if i were to do ```!isSessionInLocalStorage() || pathname !== '/register'``` then itll basically run every time the url isnt '/register' which i don't want. I only want it to check the url ONLY if the first IF is true